parlai/core/torch_ranker_agent.py

```diff
@@ -285,7 +290,7 @@ def share(self):
 
     def update_params(self):
         """Do optim step and clip gradients if needed."""
-        if self.clip > 0:
+        if hasattr(self, 'clip') and self.clip > 0:
```

Comment:
simpler: `if self.get('clip', 0) > 0:`

Reply:
Should be `self.get('gradient_clip', 0)` I believe.

Reply:
Not sure where this self.get comes from, TRA is not a dict ?